consensus/codec.go

```diff
@@ -1,15 +1,386 @@
 package consensus
```

Comment:
Can we rename this file `messages.go` or something? Codec has a certain Amino-flavor. :)